pkg/config/loader.go

```diff
@@ -49,15 +50,21 @@ type Loader struct {
 	// The Kubernetes client that will be used to load Kubernetes objects:
 	client kubernetes.Interface
 
-	// The names of the configuration files, in the order that they should be loaded:
+	// The names of the configuration directories and files, in the order that they should be loaded:
 	files []string
 
+	// The names of loaded configuration files:
+	LoadedFiles []string
+
 	// The configuration object that is being populated:
 	config *Config
 
 	// The codec that will be used to convert the rules specified in the configuration file into the
 	// types used internally.
 	codec runtime.Codec
+
+	// The config file change watcher.
+	ConfigFilesObserver observer.Observer
```

Comment:
Why does this need to be public?

Not sure how the observer package works, but shouldn't this be a pointer? From what I see in the observer project `Observer` has several fields, and using it directly means that with every call it will it all be copied, which is heavy.

Reply:
looking into it